geoh5py/ui_json/__init__.py

```diff
@@ -18,6 +18,67 @@
 # pylint: disable=unused-import
 # flake8: noqa
 
+from __future__ import annotations
+
+from typing import Any
+
 from .input_file import InputFile
 from .utils import monitored_directory_copy
 from .validation import InputValidation
+
+
+class SetDict(dict):
+    """
+    A dictionary that stores a collection of unique values.
+
+    :Note: Setting replaces, but an update results in the union of
+         any existing and update sets.
+    """
+
+    def __init__(self, **kwargs):
+        self.update(kwargs)
+
+    def update(self, data: dict[str, Any]):  # type: ignore
+        """Update keys and associated sets."""
+
+        for key, value in data.items():
+            if key in self.__dict__:
+                self.__dict__[key].update(self._make_set(value))
+            else:
+                self[key] = value
+
+    def _make_set(self, value) -> set:

```

Comment:
This looks like it could be static

Reply:
Perhaps, but I wouldn't open this up unless there's a good reason.  This just invites coupling between classes.

Reply:
...If you think it's useful elsewhere I would pull it out as a utility instead